### PR TITLE
First attempt at a click anywhere

### PR DESCRIPTION
**Pull request by sleighsoft**
I unfortunately messed up the repository and had to recreate my PR.

This is a reference to https://github.com/plotly/plotly.js/pull/5138

### CODE REVIEW

**Comment by sleighsoft** [comment]
@nicolaskruchten Any idea why the following happens when running `python setup.py updateplotlyjsdev --devbranch pull/5416`

```
    ERROR in ./node_modules/plotly.js/dist/plotly.js
    Module build failed: SyntaxError: Unexpected character '�' (5879:32)
```

**Comment by sleighsoft** [comment]
And when running `npm run test-jasmine -- test\jasmine\tests\click_test.js 
--FF --nowatch` in the `plotly.js` repo 18 tests fail.

**Comment by sleighsoft** [comment]
Also, is there a good place to add a demo?

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="../plotly.js/dist/plotly.js"></script>
</head>

<body>
    <div id="tester" style="width:600px;height:250px;"></div>
    <script>
        var TESTER = document.getElementById('tester');
        Plotly.newPlot(TESTER, [{
            x: [1, 2, 3, 4],
            y: [4, 1, 3, 5],
            mode: "lines+markers",
            marker: {size: 12},
        }], {
            margin: { t: 0 },
            clickmode: "event+anywhere"
        });
        TESTER.on('plotly_click', function(clickData) {
            var x = clickData["points"][0]["x"];
            Plotly.relayout(TESTER, {
                shapes: [
                {
                    type: "line",
                    x0: x,
                    y0: 0,
                    x1: x,
                    y1: 1,
                    yref: "paper",
                    line: {color: "RoyalBlue", width:3},
                }
                ]
            });
        });
    </script>
</body>
</html>
```

**Comment by archmoj** [comment]
> @nicolaskruchten Any idea why the following happens when running `python setup.py updateplotlyjsdev --devbranch pull/5416`
> 
> ```
>     ERROR in ./node_modules/plotly.js/dist/plotly.js
>     Module build failed: SyntaxError: Unexpected character '�' (5879:32)
> ```
> 
> It looks like the build artifact has a weird encoding? (I am on Windows 10)
> 

Similar to #5382 
Possibly would be fixed by d3-geo cc: #5112

**Comment by archmoj** [comment]
> I unfortunately messed up the repository and had to recreate my PR.
> 

Thanks for the new PR.
In order to be able to sync the master branch of your fork with upstream master, next time I suggest you open PRs from another branch on your fork (not the master).

File: src/components/fx/click.js [hunk 1]
```diff
@@ -22,14 +22,29 @@ module.exports = function click(gd, evt, subplot) {
...
+            if(annotationsDone && annotationsDone.then) {
+                annotationsDone.then(function() { emitClick(data); });
+            } else emitClick(data);
+        }
 
         // why do we get a double event without this???
         if(evt.stopImmediatePropagation) evt.stopImmediatePropagation();
     }
-};
+};
```
Comment (archmoj) [inline src/components/fx/click.js:50]: Please add a new line to the end of this file.
To catch syntax errors you could run
npm run test-syntax && npm run lint

**Comment by archmoj** [comment]
> And when running `npm run test-jasmine -- test\jasmine\tests\click_test.js --FF --nowatch` in the `plotly.js` repo 18 tests fail if I select the latest commit (without my changes).

Thanks for running them locally.
Some tests are not stable and fail locally; but they all passed on the CI.

**Comment by sleighsoft** [comment]
> > I unfortunately messed up the repository and had to recreate my PR.
> 
> Thanks for the new PR.
> In order to be able to sync the master branch of your fork with upstream master, next time I suggest you open PRs from another branch on your fork (not the master).

Thanks for pointing that out. I have been in a hurry and must have missed that. Should I open another new PR to be able to stay in sync with the plotly:master? I guess I've to wait for your geo fix to pass to be able to run `python setup.py updateplotlyjsdev`

**Comment by archmoj** [comment]
The bundles are fixed on master after #5426.
So you don't have to wait for the geo PR.

**Comment by sleighsoft** [comment]
Thank you @archmoj. Will pull those in.

I have a question regarding testing. I have look through the code and did not find how you are testing the different `clickmode` flags under `test\jasmine\tests\click_test.js`. Could you give me some guidance/help on how to implement a proper test for it, please.

**Pull request closed without merging**